app/models/exchange.rb

```diff
@@ -48,8 +48,9 @@ class Exchange < ActiveRecord::Base
       select('DISTINCT exchanges.*')
   }
   scope :with_product, lambda { |product|
+    exchange_variant_ids = product.variants_including_master.select('spree_variants.id').map(&:id)
```

Comment:
same as above, I think .select.map is equivalent to .pluck

Reply:
but in this case it would also be better to just use select because it can go as a subquery in the following in clause.